### ISSUE TITLE
add_custom_target cannot create target "mlpack_arma_config" because another target with the same name already exists

### ISSUE THREAD

**Issue by gopalkb** [mlpack/mlpack#2906]
I have been following [this](https://keon.github.io/mlpack-on-windows/) and [this](https://www.mlpack.org/doc/stable/doxygen/build_windows.html) for setting up mlpack on Windows 10 using Visual Studio 2019. While I have been able to compile everything till Armadillo binaries, when compiling the mlpack using cmake is turning out to be really frustrating now. After going through a lot of errors, I thought of coming up with a CMakeLists file which is just able to compile without errors, however, it doesn't create the mlpack binaries (so useless at the end). What is frustrating is that the process is complaining for src/mlpack directory not being present. Then when I create it manually, it starts complaining about the CMakeLists file not being present. Do that, and then it complains about a CMake directory and some other cmake files inside it. Even after that, it starts to do the whole process again recursively. 

One more thing which is not working is the dependency on pkg-config. In all the methods mentioned [here](https://stackoverflow.com/questions/1710922/how-to-install-pkg-config-in-windows) for installing it on windows, none of them have been able to explain how to install the pre-compiled binaries. So, this whole process of setting up mlpack is turning out to be a nightmare and a complete waste of effort just for getting started.

So, my question is this - Is it any working step-by-step guide for windows 10 or is there any minimal working CMakeLists.txt file for getting this to work. I have attached the CMakeLists.txt just for reference.
[CMakeLists.txt](https://github.com/mlpack/mlpack/files/6273044/CMakeLists.txt)

Thanks.

**Comment by rcurtin** [comment]
Hi @gopalkb; sorry that you have been having a frustrating time.  Honestly, compiling complicated C++ libraries on Windows can often be difficult.  I am not sure if it's realistic in your setup, but if you can, you may have an easier time if you can switch to Linux (or similar).  Personally, I am not very much of a Windows expert, but I can try and help you debug.

I would really not suggest using a custom-written CMake configuration unless you really know what you are doing.  mlpack's CMake configuration is quite complicated and it could be very difficult to replicate the functionality.

More likely, your route to success should be to set each of the CMake configuration variables correctly.  I could try and provide a bit more advice, but you haven't provided any actual error output.  I believe many others have compiled mlpack successfully on Windows 10 systems, so it should be possible.

Do you really need pkg-config?  That doesn't seem very useful in a Windows environment.

**Comment by gopalkb** [comment]
Hi @rcurtin Thanks for your response. I don't seem to have much knowledge on the build process using CMake, so it is something I will need to spend some time on. Also, I am not sure why or if I need pkg-config in my case, but the thing I noticed is when I commented the code in the CMake file related to pkg-config, I didn't get any errors but the mlpack binaries were not there in the directory. I was under impression that the step-by-step guide will have the necessary CMake files for the minimal workable setup.  
As far as the error is concerned, I will try to build again and will post the errors.
